### PR TITLE
{bio}[GCC/11.2.0] Regenie v3.1.1, BGEN-enkre v1.1.7, Boost v1.55.0

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description = """Regenie is a C++ program for whole genome regression modelling of large genome-wide 
+association studies.  It is developed and supported by a team of scientists at the Regeneron Genetics Center."""
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = ['https://github.com/rgcgithub/regenie/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = ['%(namelower)s-%(version)s_Makefile.patch']
+
+checksums = []
```
Comment (verdurin) [inline easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb:21]: Please add the missing checksums.

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Contribution from the NIHR Biomedical Research Centre
+# Guy's and St Thomas' NHS Foundation Trust and King's College London
+# uploaded by J. Sassmannshausen
+# we recommend to use --download-timeout=1000 when fetching the files
+
+easyblock = 'MakeCp'
+
+name = 'Regenie'
+version = '3.1.1'
```
[the remark this answers is not in the document]
Comment (fizwit) [reply easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb:9]: Just rebuild with 31.2 on my local system without errors. Will update Pr

File: easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb (hunk 1 above)
Comment (fizwit) [reply easybuild/easyconfigs/r/Regenie/Regenie-3.1.1-GCC-11.2.0.eb:21]: ok

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
centos7.lan - Linux CentOS Linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/07440ab1141b3ff57e98626e0787e63f for a full test report.

**Comment by fizwit** [comment]
You pinged me?

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/ddb8098acf1033f348aa4c4a0c20d82a for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
centos7.lan - Linux CentOS Linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/eebc193b00796e7cf400bdb218446d6d for a full test report.

**Comment by verdurin** [comment]
@fizwit yes, I think they have regenerated the tarball in some way, so the checksum has changed (date is now 2022-09-10). Could you check, please?

**Comment by fizwit** [comment]
The checksum has not changed.  What checksum do you have?  I'll add it to the list.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/4c04c68bbcc8a17daea23f0936539be5 for a full test report.

**Comment by verdurin** [comment]
@fizwit here's what I see:
```
== 2024-07-04 08:18:04,437 filetools.py:1307 DEBUG Computed sha256 checksum for /eb/maint/sources/b/BGEN-enkre/1.1.7.tgz: b922ac22c1c0e365d0de6054f6ce2ad911bc81db5bcd8ca915bae750f57bd0a7 (correct checksum: 8173d614ae629f00c84c0e6edd7d16f56a61af2e5a8717e8fe0167f530a6aabb)
== 2024-07-04 08:18:04,439 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:126 in __init__): Checksum verification for /eb/maint/sources/b/BGEN-enkre/1.1.7.tgz using 8173d614ae629f00c84c0e6edd7d16f56a61af2e5a8717e8fe0167f530a6aabb failed. (at easybuild/framework/easyblock.py:2468 in checksum_step)
```

**Comment by verdurin** [comment]
That's the checksum I see when I download it on my laptop, too i.e.
`b922ac22c1c0e365d0de6054f6ce2ad911bc81db5bcd8ca915bae750f57bd0a7`

**Comment by fizwit** [comment]
This has happened to me before. The EU version is different from North America's. I am not able to reproduce the checksum that you see. Interestingly, there are four different tarballs of 1.1.7 to choose from.
```
6476b077af6c8e98e85fd7e09f58cb3fdf143ff91850c984248fd4dc2d74a8c3  v1.1.7.tar.gz
6476b077af6c8e98e85fd7e09f58cb3fdf143ff91850c984248fd4dc2d74a8c3  v1.1.7.tgz
dc1ba4c94bc634a6ac77690031c11f947a052c633dee3914fc61d063437faac2  1.1.7.tar.gz
8173d614ae629f00c84c0e6edd7d16f56a61af2e5a8717e8fe0167f530a6aabb  1.1.7.tgz
```

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 23 out of 23 (3 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/eb1501049b4129498ac0558f69130099 for a full test report.

**Comment by fizwit** [comment]
@verdurin,  @Micket has requested that we compare downloads. It would be nice if there was a shared location where we could compare source downloads.

**Comment by Micket** [comment]
As far as i know, you are the only person who managed to download all the different versions, everyone else is getting
`8173d614ae629f00c84c0e6edd7d16f56a61af2e5a8717e8fe0167f530a6aabb  1.1.7.tgz`
so, you can just unpack the ones you have and diff? I simply suspect that it's either
1. partial download and only the last version works at all, so the checksum is doing its job
2. autogenerated tarball with updated timestamps, which makes checksums unreliable at all.

**Comment by fizwit** [comment]
The difference is the name of the top directory. 
```
diff --brief 1_1_7_tar_gz/1.1.7 1_1_7_tgz/1.1.7.tgz/
```

File: easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+developers writing their own implementations of the BGEN format.
+Please cite:
+Band, G. and Marchini, J., "BGEN: a binary file format for imputed genotype and haplotype data", 
+bioArxiv 308296; doi: https://doi.org/10.1101/308296
+"""
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = ['https://code.enkre.net/bgen/tarball/release/']
+sources = ['%(version)s.tgz']
```
Comment (Micket) [inline easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb:24]: either replace with git config checkout or replace release with v%(version)s

Comment (fizwit) [reply easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb]: done

Comment (Micket) [reply easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb]: No it's not, you haven't fixed the source url at all, it still says "release" and instead you changed the source filename which i explained above makes no difference.

Comment (fizwit) [reply easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb]: Oh, OK.  Replace the word "release" in the URL.  Done.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/01fd2e9cc8a9cc80d83aa5d016a7fecb for a full test report.

**Comment by verdurin** [comment]
@Micket looks okay to me now - happy for it to be merged?

File: easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb (hunk 3 above)
Comment (verdurin) [reply easybuild/easyconfigs/b/BGEN-enkre/BGEN-enkre-1.1.7-GCC-11.2.0.eb]: Looks okay to me now.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
@boegelbot please test @ jsc-zen3

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @fizwit!

**Comment by verdurin** [comment]
Going in, thanks @fizwit!

**Pull request merged**